lib/src/date-input/DateInput.tsx

```diff
@@ -36,28 +41,35 @@ const DxcDateInput = React.forwardRef<RefType, DateInputPropsType>(
       tabIndex,
     },
     ref
-  ) => {
+  ): JSX.Element => {
     const [innerValue, setInnerValue] = useState(defaultValue);
     const [isOpen, setIsOpen] = useState(false);
-    const [anchorEl, setAnchorEl] = useState(null);
+    const [calendarId] = useState(`date-picker-${uuidv4()}`);
+
+    useLayoutEffect(() => {
+      if (!disabled) {
+        let actionButtonRef;
+        if (clearable) {
+          actionButtonRef = refDate?.current.getElementsByTagName("button")[1];
```

Comment:
The useLayoutEffect is placed before the `refDate` declaration, although it hasn't been declared. I know is inside a useLayoutEffect which may be executed later on but it's a bit confusing from a code-understanding perspective.